### PR TITLE
Switch away from `rouille` to `axum` in telemetry and REST servers

### PR DESCRIPTION
**Pull request by romac**
Closes: #1658
Closes: #1478
Closes: #3194

## Description

`rouille` hasn't been updated in a while, and brings with it obsolete dependencies which are either unmaintained or who have CVEs open against them, which is security liability.

This PR switches `rouille` out for `axum`, which cuts down on the number of dependencies of the relayer since we already depend on `axum` via `tonic`.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,8 +14,6 @@ members = [
     "tools/check-guide",
     "tools/query-events",
 ]
-
-[patch.crates-io]
```
Comment (ljoss17) [inline Cargo.toml:18]: Is this removed for a reason or is it just for aesthetic reasons ?

**Review by ljoss17 — APPROVED** [review]
Looks good to me!

File: Cargo.toml (hunk 1 above)
Comment (romac) [reply Cargo.toml:18]: Not sure how it got removed but there is no real reason to either leave it or keep it.

Comment (ljoss17) [reply Cargo.toml:18]: Ok, with the commented block underneath I have a slight preference if it's kept then. Without the `[patch.crates-io]` the block might be confusing.
Or the commented block could also be removed if it seems better that way

Comment (romac) [reply Cargo.toml:18]: Oh right, didn't realize there was a comment block below, will re-add

**Pull request merged**